### ISSUE TITLE
关于训练

### ISSUE THREAD

**Issue by ylc2580** [cfzd/Ultra-Fast-Lane-Detection#25]
你好，我在训练时候一直卡在这里了，请问你训练过程中载入会很慢吗？是我哪里出问题了吗？

(ylc) root@dell-PowerEdge-T640:/data/ylc/lane_detect/Ultra-Fast-Lane-Detection-master# sh launch_training.sh
[2020/07/08 14:46:20] start training...
Config (path: configs/culane_ylc.py): {'dataset': 'CULane_ylc', 'data_root': './data/CULane/driver_37_30/', 'epoch': 50, 'batch_size': 32, 'optimizer': 'SGD', 'learning_rate': 0.1, 'weight_decay': 0.0001, 'momentum': 0.9, 'scheduler': 'multi', 'steps': [25, 38], 'gamma': 0.1, 'warmup': 'linear', 'warmup_iters': 695, 'use_aux': True, 'griding_num': 200, 'backbone': '18', 'sim_loss_w': 0.0, 'shp_loss_w': 0.0, 'note': '', 'log_path': './', 'finetune': './model_download/culane_18.pth', 'resume': None, 'test_model': None, 'test_work_dir': None}
finetune from  ./model_download/culane_18.pth
105

我的训练步骤如下：
（1）由于网速问题，自己只下载了部分的culane数据集，并将对应的图片和二值图的名称保存为txt文件
（2）在culane.py中修改了fintune为下载的模型的路径；
（3）修改 log_path 为'./'，修改data_root为自己放置数据所在的路径。
（4）修改dataloader中与数据相关的路径
（5）训练

**Comment by ylc2580** [comment]
你好，请问咱们train.py中的 cp_projects(work_dir)  这句代码的含义是什么呢？一直是卡在这里的，注释掉之后，代码可以正常训练了

**Comment by cfzd** [comment]
@ylc2580 
It might be caused by one of our features, auto backup. For every time of training, the code will automatically copy all the code to the `work_dir` according to `.gitignore`. You can see the code in [here](https://github.com/cfzd/Ultra-Fast-Lane-Detection/blob/65f0231dccc0070d97c6f3aa159507ad86a5488a/utils/common.py#L74-L90).


This tool is useful for debugging and comparison between different experiments. But if you put many things to the root directory of our project and these files can not be filtered by `.gitignore` file, ***these files will be copied to the*** `work_dir`. This kind of copying will block the execution of the training.

You can delete the extra files in the root directory of this project. You can also turn off the auto backup function.

**Comment by ylc2580** [comment]
> @ylc2580
> It might be caused by one of our features, auto backup. For every time of training, the code will automatically copy all the code to the `work_dir` according to `.gitignore`. You can see the code in [here](https://github.com/cfzd/Ultra-Fast-Lane-Detection/blob/65f0231dccc0070d97c6f3aa159507ad86a5488a/utils/common.py#L74-L90).
> 
> This tool is useful for debugging and comparison between different experiments. But if you put many things to the root directory of our project and these files can not be filtered by `.gitignore` file, _**these files will be copied to the**_ `work_dir`. This kind of copying will block the execution of the training.
> 
> You can delete the extra files in the root directory of this project. You can also turn off the auto backup function.

你好，还有就是想问一下，resume和fintune为none训练出来的模型有350多兆，这和咱们的差异好大，可能的原因是什么呢？
非常感谢。

**Comment by cfzd** [comment]
@ylc2580 
As you may see here:
https://github.com/cfzd/Ultra-Fast-Lane-Detection/blob/65f0231dccc0070d97c6f3aa159507ad86a5488a/utils/common.py#L63-L70
There are some state weights for the optimizer. Also, the provided trained model does not contain the weights for auxiliary segmentation branch, while the one saved in training contains the weights for auxiliary segmentation branch.
